src/components/TextInput/index.tsx

```diff
@@ -39,49 +43,41 @@ function TextInput(props) {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
-    const isLabeledMultiline = Boolean(props.label.length) && props.multiline;
+    const isLabeledMultiline = Boolean(label?.length) && multiline;
     const labelAnimationStyle = {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
         '--active-label-translate-y': `${styleConst.ACTIVE_LABEL_TRANSLATE_Y}px`,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
         '--active-label-scale': `${styleConst.ACTIVE_LABEL_SCALE}`,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
         '--label-transition-duration': `${styleConst.LABEL_ANIMATION_DURATION}ms`,
     };
 
     return (
         <BaseTextInput
             // eslint-disable-next-line react/jsx-props-no-spreading
             {...props}
-            innerRef={(el) => {
+            ref={(el) => {
+                // @ts-expect-error We need to reassign this ref to the input ref
                 textInputRef.current = el;
```

Comment:
As an option without `@ts-expect-error`
```suggestion
                // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
                (textInputRef.current as HTMLFormElement) = el;
```